msglibrary/src/test/java/ingsw/group1/msglibrary/SMSPeerTest.java

```diff
@@ -18,113 +15,74 @@
  * @author Giorgia Bortoletti
  * @author Riccardo De Zen
  */
-@Config(sdk = Build.VERSION_CODES.P)
-@RunWith(RobolectricTestRunner.class)
+@RunWith(Parameterized.class)
 public class SMSPeerTest {
 
-    private SMSPeer peer;
-    private static final String EX_VALID_ADDRESS = "+39892424";
-
-    @Test
-    public void isAddressValidNull(){
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NOT_PHONE_NUMBER,
-                SMSPeer.isAddressValid(null)
-        );
+    public static final int TEST_RUNS = 100;
+    private final int run;
+
+    /**
+     * @return parameters for the test. Just the index of the test run.
+     */
+    @Parameterized.Parameters
+    public static Object[] data() {
+        Integer[] params = new Integer[TEST_RUNS];
+        for(int i = 0; i < TEST_RUNS; i++)
+            params[i] = i;
+        return params;
     }
 
-    @Test
-    public void isAddressValidEmptyOrBlank(){
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NOT_PHONE_NUMBER,
-                SMSPeer.isAddressValid("")
-        );
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NOT_PHONE_NUMBER,
-                SMSPeer.isAddressValid("     ")
-        );
-    }
-
-    @Test
-    public void isAddressValidNaN(){
-        String notANumber = "Wassup";
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NOT_PHONE_NUMBER,
-                SMSPeer.isAddressValid(notANumber)
-        );
-    }
-
-    @Test
-    public void isAddressValidNoCountryCode(){
-        String noCC = "911";
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NO_COUNTRY_CODE,
-                SMSPeer.isAddressValid(noCC)
-        );
+    /**
+     * @param run test index.
+     */
+    public SMSPeerTest(int run){
+        this.run = run;
     }
 
-    @Test
-    public void isAddressValidTooLong(){
-        String longAddress = "+11111111111111111111111111";
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_TOO_LONG,
-                SMSPeer.isAddressValid(longAddress)
-        );
-    }
-
-    @Test
-    public void isAddressValidTooShort(){
-        String shortAddress = "+10";
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_TOO_SHORT,
-                SMSPeer.isAddressValid(shortAddress)
-        );
-    }
-
-    @Test
-    public void isAddressValidIsValid(){
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_VALID,
-                SMSPeer.isAddressValid(EX_VALID_ADDRESS)
-        );
-    }
+    private static final RandomSMSPeerGenerator GENERATOR = new RandomSMSPeerGenerator();
+    private SMSPeer peer;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-message-library/pullRequest?prid=4741620)